src/Widgets/HeaderBar.vala

```diff
@@ -47,38 +44,25 @@ public class Akira.Widgets.HeaderBar : Gtk.Box {
     public signal void new_window ();
 
     public HeaderBar () {
-        Object (orientation: Gtk.Orientation.VERTICAL, toggled: true);
+        Object (toggled: true);
     }
 
     construct {
-        headerbar = new Gtk.HeaderBar ();
-
-        headerbar.set_title (APP_NAME);
-        headerbar.set_show_close_button (true);
-
-        pack_start (headerbar, true, true, 0);
-
-        toolbar = new Gtk.Box (Gtk.Orientation.HORIZONTAL, 0);
-        toolbar.margin_start = 15;
-        toolbar.margin_end = 15;
-        toolbar.margin_top = 5;
-        toolbar.margin_bottom = 15;
+        set_title (APP_NAME);
+        set_show_close_button (true);
 
-        Gtk.Menu menu_items = new Gtk.Menu ();
+        var menu_items = new Gtk.Menu ();
         menu_items.add (new Gtk.MenuItem.with_label(_("Open")));
         menu_items.add (new Gtk.MenuItem.with_label(_("Save")));
         menu_items.add (new Gtk.MenuItem.with_label(_("Save As")));
         menu_items.add (new Gtk.SeparatorMenuItem ());
         menu_items.add (new Gtk.MenuItem.with_label(_("Quit")));
         menu_items.show_all ();
 
-        menu = new Akira.Partials.MenuButton ("folder", _("Menu"), _("Open Menu"));
+        menu = new Akira.Partials.MenuButton ("document-open", _("Menu"), _("Open Menu"));
```

Comment:
You want to be careful to use icon names from actions for tool items. The semantic name of the icon is important and it will keep your app from breaking if this icon ever changes

Reply:
Make sense, I don't really like the gear as a "Menu Open" image, but it's better to maintain consistency with the naming. I can always create a custom icon if I want to in the future.

Reply:
Oh I see, the document-open is the same as the folder, but better for the semantic name. Thank :D